service/hub/main.go

```diff
@@ -0,0 +1,181 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"flag"
+	"math/rand"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/cesanta/glog"
+	"golang.org/x/crypto/acme/autocert"
+	fsnotify "gopkg.in/fsnotify.v1"
+
+	"github.com/labring/service/hub/server"
+)
+
+type RestartableServer struct {
+	configFile string
+	authServer *server.AuthServer
+	hs         *http.Server
+}
+
+func ServeOnce(c *server.Config, cf string) (*server.AuthServer, *http.Server) {
+	as, err := server.NewAuthServer(c)
+	if err != nil {
+		glog.Exitf("Failed to create auth server: %s", err)
+	}
+
+	tlsConfig := &tls.Config{
+		MinVersion:               tls.VersionTLS13,
+		PreferServerCipherSuites: true,
+	}
+	if c.Server.HSTS {
```

Comment:
Since we are deploy this service under k8s cluster, we do not need implement https/lets encrypt relative code ourselves, just using `ingress` and/or `apisix` to do https offload, and make our own code simpler.